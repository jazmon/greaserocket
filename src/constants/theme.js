// @flow
import color from 'color';

const theme = {
  DARK_PRIMARY_COLOR: '#388E3C',
  DEFAULT_PRIMARY_COLOR: '#4CAF50',
  LIGHT_PRIMARY_COLOR: '#C8E6C9',
  TEXT_PRIMARY_COLOR: '#FFFFFF',
  ACCENT_COLOR: '#FFC107',
  PRIMARY_TEXT_COLOR: 'rgba(0, 0, 0, 0.87)',
  SECONDAY_TEXT_COLOR: 'rgba(0, 0, 0, 0.54)',
  DIVIDER_COLOR: 'rgba(0, 0, 0, 0.12)',
  DISABLED_TEXT_COLOR: 'rgba(0, 0, 0, 0.38)',
  WHITE: '#FFF',
  BLACK: '#000',
};

export default {
  ...theme,
  LIGHT_STATUS_BAR: color(theme.DARK_PRIMARY_COLOR).dark(),
};
