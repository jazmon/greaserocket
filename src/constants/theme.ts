// @flow
import color from 'color';

const theme = {
  colors: {
    primary: {
      normal: '#4CAF50',
      dark: '#388E3C',
      light: '#C8E6C9',
    },
    accent: {
      normal: '#FFC107',
    },
    white: '#FFFFFF',
    black: '#000000',
    transparent: 'transparent',
  },
  text: {
    colors: {
      light: '#FFFFFF',
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.54)',
      disabled: 'rgba(0, 0, 0, 0.38)',
      divider: 'rgba(0, 0, 0, 0.12)',
    },
    size: 12,
  },
};

export type ThemeType = typeof theme;

export function getStatusBarTextColor(
  backgroundColor: string = 'light-content',
): 'light-content' | 'dark-content' {
  return color(backgroundColor).dark() ? 'light-content' : 'dark-content';
}

export default theme;
