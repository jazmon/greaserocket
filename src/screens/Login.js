// @flow
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

import { login } from '../redux/modules/user';
import theme from '../constants/theme';

import type { ReduxState } from '../redux/modules';

type Props = {
  navigation: Object,
  route: Object,
  loading: boolean,
  login(): void,
  error: ?Error,
  profile: ?Auth0Profile,
  token: ?Auth0Token,
};

type State = {};

// type State = {error: ?Error, profile: ?Auth0Profile, token: ?Auth0Token};

// const initialState: State = { error: null, profile: null, token: null };

class Login extends React.Component {
  props: Props;

  componentDidMount() {
    this.props.login();
  }

  componentWillReceiveProps(nextProps: Props) {
    if (!!nextProps.token && nextProps.profile) {
      this.props.navigation.navigate('App');
    }
  }

  state: State;

  render() {
    const { profile, token, error } = this.props;
    if (profile && token) {
      return (
        <View style={styles.container}>
          <Text>Welcome {profile.name}</Text>
          <Text>Your email is: {profile.email}</Text>
        </View>
      );
    } else if (error) {
      return (
        <View style={styles.container}>
          <Text>{error}</Text>
        </View>
      );
    }
    return <View />;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: theme.WHITE,
    marginTop: 24,
    justifyContent: 'center',
    padding: 16,
  },
});

const mapState = ({ user }: ReduxState) => ({ ...user });

const mapActions = dispatch => ({ login: () => dispatch(login()) });

export const LoginComponent = Login;
export default connect(mapState, mapActions)(Login);
