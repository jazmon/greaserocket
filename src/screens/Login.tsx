// @flow
import React from 'react';
import { Text } from 'react-native';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import { Dispatch } from 'redux';

import { login } from 'redux/modules/user';
import { ThemeType } from 'constants/theme';

import { ReduxState } from 'redux/modules';
import { Maybe } from 'types';
import { Auth0Profile, Auth0Token } from 'types/auth';

type Props = {
  navigation: Object;
  route: Object;
  loading: boolean;
  login(): void;
  error: Maybe<Error>;
  profile: Maybe<Auth0Profile>;
  token: Maybe<Auth0Token>;
};

const Container = styled.View`
  flex: 1;
  flex-direction: column;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
  margin-top: 24px;
  justify-content: center;
  padding: 16px;
`;

class Login extends React.Component<Props, void> {
  componentDidMount() {
    this.props.login();
  }

  componentWillReceiveProps(nextProps: Props) {
    if (!!nextProps.token && nextProps.profile) {
      this.props.navigation.navigate('App');
    }
  }

  render() {
    const { profile, token, error } = this.props;
    if (profile && token) {
      return (
        <Container>
          <Text>
            Welcome {profile.name}
          </Text>
          <Text>
            Your email is: {profile.email}
          </Text>
        </Container>
      );
    } else if (error) {
      return (
        <Container>
          <Text>
            {error}
          </Text>
        </Container>
      );
    }
    return <Container />;
  }
}

const mapState = ({ user }: ReduxState) => ({ ...user });

const mapActions = (dispatch: Dispatch<ReduxState>) => ({
  login: () => dispatch(login()),
});

export const LoginComponent = Login;
export default connect(mapState, mapActions)(Login);
