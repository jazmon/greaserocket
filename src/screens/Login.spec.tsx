import 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'constants/theme';

import { LoginComponent as Login } from './Login';

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Login login={jest.fn} />
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
