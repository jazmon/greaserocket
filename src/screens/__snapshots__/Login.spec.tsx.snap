// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginTop": 24,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      undefined,
    ]
  }
/>
`;
